Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -176,4 +176,280 @@ void doDepositionShapeN(const Real * const xp, const Real * const yp, const Real
         );
 }
 
+// Compute shape factor and return index of leftmost cell where 
+// particle writes.
+// Specialized templates are defined below for orders 1, 2 and 3.
+template <int depos_order>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+int compute_shifted_shape_factor(Real* const sx, Real x_old, int i_new) {return 0;};
+
+// Compute shape factor for order 1.
+template <>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+int compute_shifted_shape_factor <1> (Real* const sx, Real x_old, int i_new){
+    int i = (int) x_old;
```

Comment:
let's make all of those `const`:
```suggestion
    const int i = (int) x_old;
```